Automattic/jetpack
# Social: Fix Social previews not using the post images

Fixes https://github.com/Automattic/jetpack-reach/issues/177

This change fixes the Social Previews in Jetpack because they weren't using images from the post-body.
In Calypso the Social previews are working fine.

## Proposed changes:

* Added a utility function to get the image from the post-body ed317e786764fb450fbcbc88c53de967825a32e8
* Started using the function if we have no image 0e6584d2b1f0133b4d7aea0c16c0712ded0881db

### Other information:


## Jetpack product discussion


https://github.com/Automattic/jetpack-reach/issues/177
## Does this pull request change what data or activity we track or use?


No.
## Testing instructions:






Do a sanity check in Calypso that the previews are working fine on the trunk.
* Create a post with [this](https://href.li/?https://assets.sonos.com/media-kits/hardware/era-300/era-300-hero-horizontal-white.jpg) image in it's post body, then in Calypso go to Posts, then 3-dot menu, then share, then click Preview. Check that the image loads fine.


Then in Jetpack:
* Check out this PR.
* Create a post, and attach [this](https://href.li/?https://assets.sonos.com/media-kits/hardware/era-300/era-300-hero-horizontal-white.jpg) image into the post body.
* Click on the `Social Previews` in the editor's sidebar.
* You should see the image being loaded in the previews.
* If you add a featured image, attached media, or turn on SIG, those should take precedence over the image in the post-body.
